src/newgrf_engine.cpp

```diff
@@ -1209,8 +1207,8 @@ void AlterVehicleListOrder(EngineID engine, uint target)
  */
 static int CDECL EnginePreSort(const EngineID *a, const EngineID *b)
 {
-	const EngineIDMapping *id_a = _engine_mngr.Get(*a);
-	const EngineIDMapping *id_b = _engine_mngr.Get(*b);
+	const EngineIDMapping *id_a = _engine_mngr.data() + *a;
+	const EngineIDMapping *id_b = _engine_mngr.data() + *b;
```

Comment:
```suggestion
	const EngineIDMapping *id_b = _engine_mngr.at(static_cast<size_t>(*a));
```
? + a few below

Reply:
Looking at the surrounding code, this makes more sense to me
```suggestion
	const EngineIDMapping &id_b = _engine_mngr[*b];
```

Reply:
Not really any different to mine, as long as it doesn't cause any issues - i added the cast as i wasn't sure whether the types would properly match, and used the `.at()` so it would throw in the case that the engine wasn't in `_engine_mngr`